File: windows/AudioTranscriptionSummary/Models/TranslationLanguage.cs
// TranslationLanguage.cs
// 翻訳先言語の列挙型（macOS 版と共通）

namespace AudioTranscriptionSummary.Models;

public enum TranslationLanguage
{
    Japanese,
    English,
    Chinese,
    Korean,
    French,
    German,
    Spanish
}

public static class TranslationLanguageExtensions
{
    public static string ToCode(this TranslationLanguage lang) => lang switch
    {
        TranslationLanguage.Japanese => "ja",
        TranslationLanguage.English => "en",
        TranslationLanguage.Chinese => "zh",
        TranslationLanguage.Korean => "ko",
        TranslationLanguage.French => "fr",
        TranslationLanguage.German => "de",
        TranslationLanguage.Spanish => "es",
        _ => "ja"
    };

    public static string ToDisplayName(this TranslationLanguage lang) => lang switch
    {
        TranslationLanguage.Japanese => "日本語",
        TranslationLanguage.English => "English",
        TranslationLanguage.Chinese => "中文",
        TranslationLanguage.Korean => "한국어",
        TranslationLanguage.French => "Français",
        TranslationLanguage.German => "Deutsch",
        TranslationLanguage.Spanish => "Español",
        _ => "日本語"
    };
}

File: macos/Sources/AudioTranscriptionSummary/Models/TranslationLanguage.swift
// TranslationLanguage.swift
// リアルタイム翻訳の対象言語を定義する列挙型

import Foundation

/// Amazon Translate でサポートする翻訳先言語
enum TranslationLanguage: String, CaseIterable, Identifiable, Codable {
    case japanese = "ja"
    case english = "en"
    case chinese = "zh"
    case korean = "ko"
    case french = "fr"
    case german = "de"
    case spanish = "es"

    var id: String { rawValue }

    /// UI 表示用の言語名
    var displayName: String {
        switch self {
        case .japanese: return "日本語"
        case .english: return "English"
        case .chinese: return "中文"
        case .korean: return "한국어"
        case .french: return "Français"
        case .german: return "Deutsch"
        case .spanish: return "Español"
        }
    }
}
